DNM stm32h7: allow Ethernet MAC without PHY

tober-auterion opened the pull request:
## Summary
- In some cases, an operational Ethernet MAC may have no PHY, for example when the system has a direct RMII MAC-to-MAC link.
- New config option STM32H7_NO_PHY
- With this option, PHY-specific code in the ethernet driver is not built
- This option is inherently incompatible with autonegotiation and speed and duplex settings must be compiled in

## Impact
- With STM32H7_NO_PHY, a MAC-to-MAC link without any PHY can be used.  Without STM32H7_NO_PHY, no impact.

## Testing
- Tested with STM32H7_NO_PHY with px4_fmu-v6s and an RMII MAC-to-MAC link
- Without STM32H7_NO_PHY - which is the default - it is reasonable to determine by inspection that this PR does not change the code from before the introduction of the option.

davids5 commented:
@tober-auterion - Please submit this upstream to NuttX so we can backport it. Please run tools/checkpathch.sh -f &lt;path to file> on it and indent the ifdefs.

niklaut commented:
Upstream PR: https://github.com/apache/nuttx/pull/12538

Please leave this PR open, until it is merged, so that we can use the same remote for the git submodule.

tober-auterion committed:
[BACKPORT] stm32h7: allow Ethernet MAC without PHY

- In some cases, an operational Ethernet MAC may have no PHY, for example
when the system has a direct RMII MAC-to-MAC link.
- New config option STM32H7_NO_PHY
- With this option, PHY-specific code in the ethernet driver is not built
- This option is inherently incompatible with autonegotiation and speed and
duplex settings must be compiled in

davids5 commented:
Replaced by https://github.com/PX4/NuttX/pull/314 as backport   

PX4 PR is https://github.com/PX4/PX4-Autopilot/pull/23311
